aten/src/ATen/core/CachingHostAllocator.h

```diff
@@ -669,25 +824,67 @@ struct CachingHostAllocatorImpl {
     TORCH_CHECK_NOT_IMPLEMENTED(false, "Not implemented for query_event");
   }
 
-  alignas(hardware_destructive_interference_size) std::mutex blocks_mutex_;
-  ska::flat_hash_set<B*> blocks_; // block list
-  ska::flat_hash_map<void*, B*> ptr_to_block_;
+  virtual S get_current_stream() const = 0;
```

Comment:
@guangyey I have added two new virtual methods here: get_current_stream() and stream_is_capturing().

Since Intel has added some preliminary support for XPUGraph to pytorch, I went ahead and implemented these for you.